service/frontend/http_api_server.go

```diff
@@ -296,18 +296,20 @@ func (h *HTTPAPIServer) errorHandler(
 }
 
 func (h *HTTPAPIServer) newMarshaler(indent string, disablePayloadShorthand bool) runtime.Marshaler {
-	marshalOpts := proxy.JSONPBMarshalerOptions{
-		Indent:                  indent,
-		DisablePayloadShorthand: disablePayloadShorthand,
-	}
-	unmarshalOpts := proxy.JSONPBUnmarshalerOptions{DisablePayloadShorthand: disablePayloadShorthand}
-	if m, err := proxy.NewJSONPBMarshaler(marshalOpts); err != nil {
-		panic(err)
-	} else if u, err := proxy.NewJSONPBUnmarshaler(unmarshalOpts); err != nil {
-		panic(err)
-	} else {
-		return proxy.NewGRPCGatewayJSONPBMarshaler(m, u)
-	}
+	return newProtoJsonMarshaler(indent)
+	// TODO: reintroduce shorthand JSON marshaling here
```

Comment:
Not in this PR but another PR that I'm working on. I will provide a new JSON marshaller as part of the Go API we'll integrate.

We are temporarily breaking the HTTP API contract _on the main branch_ because it is not currently available in production. It _will_ be fixed before this work is officially released. This has been signed off on by Chad, Yimin, and the HTTP API crew already.

I'm halfway through reimplementing this on top of the new proto library and that will be put up as a separate review.